tools/src/commands/GenerateBareApp.ts

```diff
@@ -187,13 +189,16 @@ async function symlinkPackages({
   }
 }
 
-async function updateRNVersion({ projectDir }: { projectDir: string }) {
+async function updateRNVersion({ projectDir, rnVersion }: { projectDir: string, rnVersion?: string }) {
+  if (!rnVersion) {
+    const mainPkg = require(path.resolve(EXPO_DIR, 'package.json'));
+    const rnVersionOnMain = mainPkg.resolutions['react-native'];
+    rnVersion = rnVersionOnMain
```

Comment:
It's a bad practice to reassign function arguments. It's better to move resolving the version to a separate function, like this:
```javascript
function getLocalReactNativeVersion() {
  const mainPkg = require(path.resolve(EXPO_DIR, 'package.json'));
  return mainPkg.resolutions?.['react-native'];
}
// and then
const reactNativeVersion = rnVersion || getLocalReactNativeVersion();
```